sdv/metadata/multi_table.py

```diff
@@ -13,6 +15,165 @@ def __init__(self):
         self._tables = {}
         self._relationships = []
 
+    @staticmethod
+    def _validate_missing_relationship_keys(parent_table, parent_table_name, parent_primary_key,
+                                            child_table_name, child_foreign_key):
+        missing_keys = set()
+        parent_primary_key = cast_to_iterable(parent_primary_key)
+        table_primary_keys = set(cast_to_iterable(parent_table._primary_key))
+        for key in parent_primary_key:
+            if key not in table_primary_keys:
+                missing_keys.add(key)
+
+        if missing_keys:
+            raise ValueError(
+                f'Relationship between tables ({parent_table_name}, {child_table_name}) contains '
+                f'an unknown primary key {missing_keys}.'
+            )
+
+        for key in set(cast_to_iterable(child_foreign_key)):
+            if key not in parent_table._columns:
+                missing_keys.add(key)
+
+        if missing_keys:
+            raise ValueError(
+                f'Relationship between tables ({parent_table_name}, {child_table_name}) '
+                f'contains an unknown foreign key {missing_keys}.'
+            )
+
+    @staticmethod
+    def _validate_missing_relationship_tables(parent_table_name, child_table_name, tables):
+        missing_table_name = set([parent_table_name, child_table_name]) - set(tables)
+        if missing_table_name:
+            raise ValueError(f'Relationship contains an unknown table {missing_table_name}.')
+
+    @staticmethod
+    def _validate_relationship_key_length(parent_table_name, parent_primary_key,
+                                          child_table_name, child_foreign_key):
+        pk_len = len(set(cast_to_iterable(parent_primary_key)))
+        fk_len = len(set(cast_to_iterable(child_foreign_key)))
+        if pk_len != fk_len:
+            raise ValueError(
+                f"Relationship between tables ('{parent_table_name}', '{child_table_name}') is "
+                f'invalid. Primary key has length {pk_len} but the foreign key has '
+                f'length {fk_len}.'
+            )
+
+    @staticmethod
+    def _validate_relationship_sdtypes(parent_table, parent_table_name, parent_primary_key,
+                                       child_table_name, child_foreign_key):
+        parent_columns = parent_table._columns
+        error = False
+        if isinstance(parent_primary_key, (list, tuple)):
+            for pk, fk in zip(parent_primary_key, child_foreign_key):
+                if parent_columns[pk]['sdtype'] != parent_columns[fk]['sdtype']:
+                    error = True
+                    break
+
+        else:
+            pk_sdtype = parent_columns[parent_primary_key]['sdtype']
+            fk_sdtype = parent_columns[child_foreign_key]['sdtype']
```

Comment:
if we made a set of all columns involved in the primary key and foreign key, then we could just loop through them instead of doing this if right? To me it might make sense to combine `_validate_relationship_key_length` with this method and do the following steps:
1. Convert keys to sets
2. If the lengths are different, raise an error
3. Otherwise, loop through the keys and make sure the sdtypes are the same